conda_smithy/feedstock_tokens.py

```diff
@@ -0,0 +1,569 @@
+import tempfile
+import os
+import json
+import sys
+import secrets
+from contextlib import redirect_stderr, redirect_stdout
+
+import git
+import requests
+import scrypt
+
+
+def generate_and_write_feedstock_token(user, project):
+    """Generate a feedstock token and write it to
+
+        ~/.conda-smithy/{user or org}_{repo w/o '-feedstock'}_feedstock.token
+
+    This function will fail if the token file already exists.
+    """
+    # capture stdout, stderr and suppress all exceptions so we don't
+    # spill tokens
+    failed = False
+    err_msg = None
+    with open(os.devnull, "w") as fp:
+        with redirect_stdout(fp), redirect_stderr(fp):
```

Comment:
This can be one level of indentation: `with open(os.devnull, "w") as fp, redirect_stdout(fp), redirect_stderr(fp)` I think

Reply:
Indeed it can!

Reply:
done